File: src/routes/private/chat.routes.js
import express from "express"

import * as CC from "../../controllers/chat.controllers.js"
import { validateParams } from "../../validators/miscs.js"

const router = express.Router()

router.get("/my_chats", CC.getMyChats)

router.delete(
  "/chats/:partner_username",
  ...validateParams,
  CC.deleteChat
)


export default router

File: src/domain/chat/chatRoutes/chatRoutes.go
package chatRoutes

import (
	CC "i9lyfe/src/domain/chat/chatControllers"

	"github.com/gofiber/fiber/v3"
)

func Routes(router fiber.Router) {
	router.Post("/chat_upload/authorize", CC.AuthorizeUpload)
	router.Post("/chat_upload/authorize/visual", CC.AuthorizeVisualUpload)

	router.Get("/chats", CC.GetChats)
	router.Delete("/chats/:partner_username", CC.DeleteChat)
}
